### PR TITLE
feat: dogface 2nd staking

### PR DESCRIPTION
**Pull request by p-dealwis**
Signed-off-by: Pramodya De Alwis <pramodya1@hotmail.com>

### CODE REVIEW

**Comment by p-dealwis** [comment]
@raymondfeng there seems to be an issue with the integration tests when using the Infura api.

File: src/adapters/dogface2.adapter.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import type {Dogface2ndStaking} from '../types/Dogface2ndStaking';
+import {Dogface2ndStaking__factory} from '../types/factories/Dogface2ndStaking__factory';
+
+@injectable(
+  {
+    scope: BindingScope.SINGLETON,
+  },
+  extensionFor(STAKING_ADAPTERS_EXTENSION_POINT),
+)
+export class Dogface2ndStakingContractAdapter extends BaseStakingContractAdapter {
```
Comment (raymondfeng) [inline src/adapters/dogface2.adapter.ts:19]: You need to register the adapter at https://github.com/abridged/collabland-staking-contracts/blob/master/src/component.ts#L59

[the remark this answers is not in the document]
Comment (p-dealwis) [reply src/adapters/dogface2.adapter.ts:19]: ah yes of course!

Comment (p-dealwis) [reply src/adapters/dogface2.adapter.ts:19]: I just added that in now. But there is this issue still: 

> Error: missing revert data in call exception; Transaction reverted without a reason string [ See: https://links.ethers.org/v5-errors-CALL_EXCEPTION ] (data="0x", transaction={"to":"0xB87ef86998cf8a0a75E5d0c323001a133a765537","data":"0xcfe8f4870000000000000000000000009abbf7218c65c4d22c8483b5d6be93075a3c159c00000000000000000000000079aa05963c92a2a10d96bd840eb5e73a1e675e92","accessList":null}, error={"reason":"bad response","code":"SERVER_ERROR","status":401,"headers":{"date":"Tue, 10 Jan 2023 20:20:04 GMT","content-type":"text/plain; charset=utf-8","content-length":"31","connection":"close","vary":"Accept-Encoding, Origin","www-authenticate":"Basic realm=\"Project ID is required in the URL\"","x-content-type-options":"nosniff"},"body":"project id required in the url\n","requestBody":"{\"method\":\"eth_call\",\"params\":[{\"to\":\"0xb87ef86998cf8a0a75e5d0c323001a133a765537\",\"data\":\"0xcfe8f4870000000000000000000000009abbf7218c65c4d22c8483b5d6be93075a3c159c00000000000000000000000079aa05963c92a2a10d96bd840eb5e73a1e675e92\"},\"latest\"],\"id\":42,\"jsonrpc\":\"2.0\"}","requestMethod":"POST","url":"https://mainnet.infura.io/v3/"}, code=CALL_EXCEPTION, version=providers/5.7.1)

When looking at the error it shows a 401 status error and the body of the error shows this message: `project id required in the url`

[the remark this answers is not in the document]
Comment (p-dealwis) [reply src/adapters/dogface2.adapter.ts:19]: So they are 2 different staking contracts, one for each asset.

Comment (raymondfeng) [reply src/adapters/dogface2.adapter.ts:19]: The PR won't be able to see our CI secrets for the infura project id.

Comment (p-dealwis) [reply src/adapters/dogface2.adapter.ts:19]: Anyway to make it work so that the test can pass? I have registered the adapter in the component file now.

**Pull request merged**